File: THCS/2023_2024/HaNoi_2324/Bai1/HOAHOC.cpp
#include <iostream>
#include <fstream>
#include <algorithm>

using namespace std;

int main() {
    // Đọc dữ liệu từ tệp HOAHOC.INP
    ifstream infile("HOAHOC.INP");
    long long A, B; // Sử dụng long long để đảm bảo đủ khả năng lưu trữ
    infile >> A >> B;
    infile.close();

    // Tính số mol của Fe3O4
    long long mFe = A / 3; // Số mol Fe3O4 phụ thuộc vào số mol Fe
    long long mO2 = B / 1; // Số mol O2 (1 mol cho mỗi 3 mol Fe3O4)
    long long result = min(mFe, mO2); // Kết quả là số mol sản phẩm Fe3O4

    // Ghi kết quả vào tệp HOAHOC.OUT
    ofstream outfile("HOAHOC.OUT");
    outfile << result << endl;
    outfile.close();

    return 0;
}

File: THCS/2023_2024/HaNoi_2324/Bai1/HOAHOC.py
def main():
    try:
        # Đọc dữ liệu từ tệp HOAHOC.INP
        with open('HOAHOC.INP', 'r') as infile:
            data = infile.read().strip()
            if not data:
                raise ValueError("Tệp HOAHOC.INP không có dữ liệu.")
            A, B = map(int, data.split())
        
        # Tính số mol của sản phẩm Fe3O4
        mol_Fe3O4_from_Fe = A // 3
        mol_Fe3O4_from_O2 = B // 2
        mol_Fe3O4 = min(mol_Fe3O4_from_Fe, mol_Fe3O4_from_O2)
        
        # Ghi kết quả vào tệp HOAHOC.OUT
        with open('HOAHOC.OUT', 'w') as outfile:
            outfile.write(str(mol_Fe3O4) + '\n')

    except FileNotFoundError:
        print("Tệp HOAHOC.INP không tồn tại.")
    except ValueError as ve:
        print(f"Lỗi: {ve}")

if __name__ == "__main__":
    main()
